File: app/src/main/java/com/jgabrielfreitas/firebase/push/MainActivity.java
package com.jgabrielfreitas.firebase.push;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import com.google.firebase.FirebaseApp;
import com.google.firebase.iid.FirebaseInstanceId;

public class MainActivity extends AppCompatActivity {

  private static final String TAG = "Main";

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
  }
}

File: app/src/main/java/com/jgabrielfreitas/firebase/push/MainActivity.kt
package com.jgabrielfreitas.firebase.push

import android.content.ClipData.newPlainText
import android.content.ClipboardManager
import butterknife.OnClick
import com.google.firebase.iid.FirebaseInstanceId
import com.jgabrielfreitas.core.activity.BaseActivity
import com.jgabrielfreitas.firebase.push.extension.toast
import com.jgabrielfreitas.layoutid.annotations.InjectLayout
import android.content.Context.CLIPBOARD_SERVICE as clipboard_service

@InjectLayout(layout = R.layout.activity_main)
class MainActivity : BaseActivity() {

  @OnClick(R.id.copyToClipboardButton) fun copyToClipboard() {

    val clipboard = getSystemService(clipboard_service) as ClipboardManager
    val clip = newPlainText("token API", FirebaseInstanceId.getInstance().token)
    clipboard.primaryClip = clip

    toast("copied: " + FirebaseInstanceId.getInstance().token)
  }
}
